codegen/type_converter.go

```diff
@@ -638,25 +671,78 @@ func (c *TypeConverter) genStructConverter(
 			var (
 				stFromPrefix = keyPrefix
 				stFromType   compile.TypeSpec
+				fromTypeName string
 			)
 			if fromField != nil {
 				stFromType = fromField.Type
 				stFromPrefix = keyPrefix + pascalCase(fromField.Name)
+
+				var err error
+				fromTypeName, err = c.getIdentifierName(stFromType)
+				if err != nil {
+					fromTypeName = ""
+				}
 			}
-			err := c.genConverterForStruct(
-				toField.Name,
-				toFieldType,
-				toField.Required,
-				stFromType,
-				fromIdentifier,
-				keyPrefix+pascalCase(toField.Name),
-				stFromPrefix,
-				indent,
-				fieldMap,
-			)
+
+			toTypeName, err := c.getIdentifierName(toFieldType)
 			if err != nil {
 				return err
 			}
+
+			if converterMethodName, ok := c.convStructMap[toFieldType.Name]; ok {
+				// the converter for this struct already exists, so just use it
+				c.append(indent, "out.", keyPrefix+pascalCase(toField.Name), " = ", converterMethodName, "(", fromIdentifier, ")")
+			} else if fromTypeName != "" {
+				// generate a converter for this scope
```

Comment:
what's the purpose for scope use here

Reply:
Originally it was to avoid name conflicts, but in hindsight that probably won't be an issue, I can try removing them